book/analysis/big-o-notation.md

```diff
@@ -75,15 +77,16 @@ of the problem. For these kinds of algorithms, we need to characterize their
 performances as *worst case*, *best case*, or *average case*.
 
 The worst case performance refers to a particular data set where the algorithm
-performs especially poorly, whereas a different data set might have amazing
-performance. In most cases, however, the algorithm performs somewhere in between
-these two extremes, in what we kindly call the average case. Understanding these
-distinctions can help prevent any one particular case from misleading us.
+performs especially poorly, while the best case performance, as you can
+probably infer, refers to a data set that the algorithm can process extremely
+fast. In most cases, however, the algorithm performs somewhere in between
```

Comment:
@aleksharma12 just noticed this now, but it's a little misleading to say that the average case happens "in most cases". That would be the mode rather than the average. It's possible that no actual execution is the average case, or that one case far exceeds the others. For instance, consider appending to a list with an underlying implementation of an array... once the array is full, we need to allocate a new (larger) array, copy all of the existing values, and deallocate the previous array. As such for an array of size N, in most cases the append is O(1) and in 1/N cases the append is O(N). By making the new array three times larger than the previous array, we end up with _average_ running time of O(1) again, but only because the cost of the resize can be amortized fairly over the subsequent appends. If we picked a smaller new array size, it would not average out to O(1) but in "most cases" it still would.